src/main/java/edu/ucsb/cs156/courses/documents/CoursePage.java

```diff
@@ -0,0 +1,79 @@
+package edu.ucsb.cs156.courses.documents;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Slf4j
+public class CoursePage {
+    private int pageNumber;
+    private int pageSize;
+    private int total;
+    private List<Course> classes;
+
+    /**
+     * Create a CoursePage object from json representation
+     * 
+     * @param json String of json returned by API endpoint {@code /classes/search}
+     * @return a new CoursePage object
+     * @see <a href=
+     *      "https://developer.ucsb.edu/content/academic-curriculums">https://developer.ucsb.edu/content/academic-curriculums</a>
+     */
+    public static CoursePage fromJSON(String json) {
```

Comment:
You have tests for these methods, so why are you excluding the class from test coverage?